src/pages/fi-receipt-vouchers/forms/ReceiptVoucherForm.js

```diff
@@ -0,0 +1,497 @@
+import { LogisticsRepository } from 'src/repositories/LogisticsRepository'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import CustomTextArea from 'src/components/Inputs/CustomTextArea'
+import { GeneralLedgerRepository } from 'src/repositories/GeneralLedgerRepository'
+import { useDocumentType } from 'src/hooks/documentReferenceBehaviors'
+import { DataSets } from 'src/resources/DataSets'
+import { getStorageData } from 'src/storage/storage'
+import { formatDateFromApi, formatDateToApi } from 'src/lib/date-helper'
+
+export default function ReceiptVoucherForm({ labels, maxAccess: access, recordId }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const [isCancelled, setIsCancelled] = useState()
+  const [isPosted, setIsPosted] = useState(false)
+  const [readOnly, setReadOnly] = useState(false)
+
+  const { documentType, maxAccess, changeDT } = useDocumentType({
+    functionId: SystemFunction.ReceiptVoucher,
+    access: access,
+    enabled: !recordId
+  })
+
+  const invalidate = useInvalidate({
+    endpointId: FinancialRepository.ReceiptVouchers.qry
+  })
+
+  const { formik } = useForm({
+    maxAccess: maxAccess,
+    enableReinitialize: false,
+    validateOnChange: true,
+    initialValues: {
+      reference: '',
+      accountId: null,
+      date: new Date(),
+      currencyId: null,
+      dtId: documentType?.dtId,
+      dgId: '',
+      amount: '',
+      baseAmount: '',
+      notes: '',
+      oDocId: '',
+      printStatus: '',
+      status: 1,
+      paymentMethod: '1',
+      cashAccountId: null,
+      plantId: null,
+      exRate: 1.0,
+      rateCalcMethod: 1,
+      contactId: null,
+      collectorId: null,
+      isVerified: true,
+      template: 1
+    },
+    validationSchema: yup.object({
+      accountId: yup.string().required(' '),
+      currencyId: yup.string().required(' '),
+      cashAccountId: yup.string().required(' '),
+      amount: yup.string().required(' '),
+      paymentMethod: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      const recordId = obj.recordId
+
+      const response = await postRequest({
+        extension: FinancialRepository.ReceiptVouchers.set,
+        record: JSON.stringify(obj)
+      })
+      if (!recordId) {
+        toast.success('Record Added Successfully')
+        formik.setFieldValue('recordId', response.recordId)
+        getData(response.recordId)
+      } else toast.success('Record Edited Successfully')
+      invalidate()
+    }
+  })
+  const editMode = !!recordId || !!formik.values.recordId
+
+  const getDefaultDT = async () => {
+    const userData = getStorageData('userData')
+
+    const _userId = userData.userId
+    try {
+      const { record: cashAccountRecord } = await getRequest({
+        extension: SystemRepository.UserDefaults.get,
+        parameters: `_userId=${_userId}&_key=cashAccountId`
```

Comment:
we can directly use `_userId=${userData.userId}...`

Reply:
Because I used this below in another repair, I had to do that.